hip_data_tools/apache/cassandra.py

```diff
@@ -32,6 +38,97 @@ def get_ssl_context(cert_path: str) -> SSLContext:
     return ssl_context
 
 
+_PYTHON_TO_CASSANDRA_DATA_TYPE_MAP = {
+    "Timestamp": "timestamp",
+    "str": "varchar",
+    "int64": "bigint",
+    "int32": "int",
+    "dict": "map",
+    "float64": "double",
+    "UUID": "UUID",
+}
+
+
+def _get_data_frame_column_types(data_frame):
+    data_frame_col_dict = {}
+    for col in data_frame:
+        data_frame_col_dict[col] = type(data_frame[col][0]).__name__
+    return data_frame_col_dict
+
+
+def convert_dataframe_columns_to_cassandra(data_frame):
+    """
+    Extracts a dictionary of column names and their cassandra data types from the dataframe
+    Args:
+        data_frame (DataFrame): the dataframe whose columns need to be extracted
+    Returns: dict
+    """
+    column_dtype = _get_data_frame_column_types(data_frame)
+    cassandra_columns = {key: _PYTHON_TO_CASSANDRA_DATA_TYPE_MAP[value] for (key, value) in
+                         column_dtype.items()}
+    return cassandra_columns
+
+
+def _pandas_factory(colnames, rows):
+    return pd.DataFrame(rows, columns=colnames)
+
+
+def _prepare_batches(prepared_statement, rows) -> list:
+    batches = []
+    itr = 0
+    batch = BatchStatement(consistency_level=ConsistencyLevel.QUORUM)
+    pbar = tqdm.tqdm(total=len(rows))
+    log.info("Preparing cassandra batches out of rows")
+    for row in rows:
+        if itr >= _CASSANDRA_BATCH_LIMIT:
+            batches.append(batch)
+            batch = BatchStatement(consistency_level=ConsistencyLevel.QUORUM)
+            itr = 0
+        batch.add(prepared_statement, row)
+        itr += 1
+        pbar.update(1)
+    batches.append(batch)
+    log.info("created %s batches out of data frame of %s rows", len(batches), len(rows))
+    return batches
+
+
+def _extract_rows_from_dataframe(dataframe):
+    (row, col) = dataframe.shape
+    pbar = tqdm.tqdm(total=row)
+    result = []
+    log.info("Converting Dataframe into List of tuples")
+    for index, row in dataframe.iterrows():
+        result.append(tuple([_clean_outgoing_values(val) for val in row]))
+        pbar.update(1)
+    return result
+
+
+def _extract_rows_from_list_of_dict(data):
+    return [tuple(dct.values()) for dct in data]
+
+
+def _clean_outgoing_values(val):
+    if isinstance(val, Timestamp):
+        return val.to_pydatetime()
+    if val is NaT:
+        return None
+    return val
+
+
+def _cql_manage_column_lists(data_frame, primary_key_column_list):
```

Comment:
Lots of logic here. This should be public

Reply:
yeah nah, its internal consistency check, making this public will only muddy the interface. I will however refactor to make the method simpler.